lib/program.rb

```diff
@@ -0,0 +1,34 @@
+require './lib/argumentsParser.rb'
+require './lib/rubyGemsApi.rb'
+
+class Program
+
+  def self.execute
+    ruby_gems_api = RubyGemsApi.new
+
+    arguments = ArgumentsParser.parse
+
+    if (arguments.name != '')
+      gem_info = ruby_gems_api.get_gem_by_name(arguments.name)
+      display_gem_info(gem_info)
+    elsif (arguments.search_term != '')
+      gems_info = ruby_gems_api.search_gems(arguments.search_term, arguments.search_limit)
+      display_gems_info(gems_info)
+    end
```

Comment:
Strange. Looks like If I give both a `name` and a `search term`, I will receive a single `gem info`. **Is this true?** I mean, this is my assumption based on this logic